### ISSUE TITLE
Enable domain based moderation on nodeinfo and other endpoint metadata

### ISSUE THREAD

**Issue by kyzh** [glitch-soc/mastodon#638]
As an instance administrator, I'd like to be able to block or silence instance based on its metadata.
So I could choose to make sure peertube instances are blocked by default or to reject pixelfed instances media files.
I see there are some information available already on what metadata is provided on fediverse.network [1]

I have no idea how to implement it myself.
That said, I'd love to be a patron and fund this feature (If glitch-soc admin would find it useful).

[1] https://fediverse.network/info#instance

* * * *

- [ x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by lanodan** [comment]
:+1: for nodeinfo (displaying it and providing it)
:-1: for it’s usage to block instances per-software

**Comment by TemporaryUsername12481** [comment]
Any update on this?  It's been implemented here: https://github.com/tootsuite/mastodon/pull/10079

**Comment by kyzh** [comment]
From my point of view, I already block user agent for some implementation of mastodon.
I would love to be able to do this from mastodon, instead of the webserver level.
I'd take the tradeoff  perfomance hit for the ease of access (eg, doing the change in the admin interface instead of SSH to change the webserver's config)

**Comment by TemporaryUsername12481** [comment]
I disagree with that feature and also blocking specific user agents.  It shouldn't be encouraged.
